api4/system.go

```diff
@@ -317,3 +319,37 @@ func getRedirectLocation(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.MapToJson(m)))
 	return
 }
+
+func pushNotificationsAck(c *Context, w http.ResponseWriter, r *http.Request) {
+	ack := model.PushNotificationAckFromJson(r.Body)
+
+	if !*c.App.Config().EmailSettings.PushNotificationsAck {
+		c.Err = model.NewAppError("pushNotificationAck", "api.push_notifications_ack.disabled.app_error", nil, "", http.StatusForbidden)
+		return
+	}
+
+	var notification *model.PushNotification
+	ack.AckAt = model.GetMillis()
+
+	if err := c.App.AckToPushProxy(ack); err != nil {
+		c.Err = model.NewAppError("pushNotificationAck", "api.push_notifications_ack.forward.app_error", nil, "", http.StatusInternalServerError)
+		return
+	}
+
+	if pluginsEnvironment := c.App.GetPluginsEnvironment(); pluginsEnvironment != nil {
+		pluginContext := c.App.PluginContext()
+		pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
+			notification = hooks.PushNotificationAck(pluginContext, ack.Id, ack.ReceivedAt, ack.AckAt)
+			return true
+		}, plugin.PushNotificationAckId)
+	}
+
+	mlog.Debug("Push notification ack received", mlog.String("ackId", ack.Id), mlog.Int64("receivedAt", ack.ReceivedAt), mlog.Int64("ackAt", ack.AckAt))
+
+	if notification != nil {
+		w.Write([]byte(notification.ToJson()))
```

Comment:
We want this to enable the send nothing in the initial notification so info doesn't travel though Apple/Google feature right?

Reply:
Yes, exactly, that is one of the features that could be implemented as a plugin using this "Ack" notification return. Other one is workaround the 4K notifications size limit, allowing us to enrich as much as we want the push notifications.